src/sentry/models/identity.py

```diff
@@ -21,20 +21,23 @@ class IdentityProvider(Model):
     """
     An IdentityProvider is an instance of a provider.
 
-    For example, we may auto generate some, such as github.com, where it's
-    ``IdentityProvider(type='github', instance='github.com')``
+    The IdentityProvider is unique on the type of provider (eg gtihub, slack,
+    google, etc) and the organization which has configured that provider for
+    it's users.
 
-    When possible the instance should be the domain used.
+    A SAML identity provide might look like this, type: onelogin, instance:
+    acme-org.onelogin.com.
     """
     __core__ = False
 
     type = models.CharField(max_length=64)
-    instance = models.CharField(max_length=64)
+    organization = FlexibleForeignKey('sentry.Organization')
```

Comment:
Ok, this migration is going to be problematic.

We need to come up with a transition plan. We can't drop and add these new columns at the same time, nor can we simply edit this unique_together, or add not nullable columns.

Reply:
[Addressed in this comment](https://github.com/getsentry/sentry/pull/6708#issuecomment-354855601)